### PR TITLE
utils(dm): fix get tables without using quote schema name

### LINKED ISSUE #5895: check-task can't handle special character in identifier
### What did you do?

https://github.com/pingcap/tiflow/blob/066e082998027f72a450a928ba6a0e7fd2cb9f68/dm/pkg/utils/db.go#L430-L432

forget to quote it.

### What did you expect to see?

_No response_

### What did you see instead?

_No response_

### Versions of the cluster

v6.1.0

### current status of DM cluster (execute `query-status <task-name>` in dmctl)

```console
(paste current status of DM cluster here)
```

### PR DESCRIPTION
**Pull request by Ehco1996**
### What problem does this PR solve?

Issue Number: close #5895

### What is changed and how it works?

* use tidb/dbutils to query db and this func  quote  the schema name

https://github.com/pingcap/tidb/blob/4ae78cdead54097679407314ce33b895e3c01d16/util/dbutil/common.go#L379

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
`fix a bug that get tables without using quote schema name`.
```

### CODE REVIEW

**Comment by Ehco1996** [comment]
/cc @lance6716 @buchuitoudegou 
ptal 🧡

**Review by buchuitoudegou — APPROVED** [review]
It seems `tidb/util` suffices a majority demand to query the database in our repo. Do we have a plan to sort them out and replace all these duplicate stuff?

**Review by lance6716 — COMMENTED** [review]
please add a test

**Comment by Ehco1996** [comment]
> It seems `tidb/util` suffices a majority demand to query the database in our repo. Do we have a plan to sort them out and replace all these duplicate stuff?

left an issue https://github.com/pingcap/tiflow/issues/5899 , we can do this when we have free time (or community developers can get involved

**Comment by Ehco1996** [comment]
> please add a test

already tested in openapi's ut and it see

https://github.com/pingcap/tiflow/blob/b9d096912f3ba392a314c19ef8fee03910f6bd86/dm/dm/master/openapi_view_test.go#L718

https://github.com/pingcap/tiflow/blob/b9d096912f3ba392a314c19ef8fee03910f6bd86/dm/tests/openapi/run.sh#L85

**Comment by lance6716** [comment]
> > please add a test
> 
> already tested in openapi's ut and it see
> 
> https://github.com/pingcap/tiflow/blob/b9d096912f3ba392a314c19ef8fee03910f6bd86/dm/dm/master/openapi_view_test.go#L718
> 
> https://github.com/pingcap/tiflow/blob/b9d096912f3ba392a314c19ef8fee03910f6bd86/dm/tests/openapi/run.sh#L85

please check what's the cause of issue, and verify it doesn't matter now.

**Comment by Ehco1996** [comment]
> > > please add a test
> > 
> > 
> > already tested in openapi's ut and it see
> > https://github.com/pingcap/tiflow/blob/b9d096912f3ba392a314c19ef8fee03910f6bd86/dm/dm/master/openapi_view_test.go#L718
> > 
> > https://github.com/pingcap/tiflow/blob/b9d096912f3ba392a314c19ef8fee03910f6bd86/dm/tests/openapi/run.sh#L85
> 
> please check what's the cause of the issue, and verify it doesn't matter now.

the func in `tidb/util` repo quote the schema name,so I think no need add a separate test ?

https://github.com/pingcap/tidb/blob/4ae78cdead54097679407314ce33b895e3c01d16/util/dbutil/common.go#L379

**Comment by lance6716** [comment]
> > > > please add a test
> > > 
> > > 
> > > already tested in openapi's ut and it see
> > > https://github.com/pingcap/tiflow/blob/b9d096912f3ba392a314c19ef8fee03910f6bd86/dm/dm/master/openapi_view_test.go#L718
> > > 
> > > https://github.com/pingcap/tiflow/blob/b9d096912f3ba392a314c19ef8fee03910f6bd86/dm/tests/openapi/run.sh#L85
> > 
> > 
> > please check what's the cause of the issue, and verify it doesn't matter now.
> 
> the func in `tidb/util` repo quote the schema name,so I think no need add a separate test ?
> 
> https://github.com/pingcap/tidb/blob/4ae78cdead54097679407314ce33b895e3c01d16/util/dbutil/common.go#L379

All tests in this repo can detect the bug. If you don't change them we may regress in future.

**Comment by lance6716** [comment]
the issue is also about `check-task`, please add a test for it.

**Comment by Ehco1996** [comment]
> the issue is also about `check-task`, please add a test for it.

added in check-task it https://github.com/pingcap/tiflow/pull/5896/commits/cbe70fa55a73f3cca8da93425bc315233800ca6c

**Comment by lance6716** [comment]
> > the issue is also about `check-task`, please add a test for it.
> 
> added in check-task it [cbe70fa](https://github.com/pingcap/tiflow/commit/cbe70fa55a73f3cca8da93425bc315233800ca6c)

this test can't find the problem without this pr.

**Comment by Ehco1996** [comment]
> > > the issue is also about `check-task`, please add a test for it.
> > 
> > 
> > added in check-task it [cbe70fa](https://github.com/pingcap/tiflow/commit/cbe70fa55a73f3cca8da93425bc315233800ca6c)
> 
> this test can't find the problem without this pr.

thanks, I was a little busy dizzy yesterday 😵
fixed in https://github.com/pingcap/tiflow/pull/5896/commits/d982b51758dd007b68e44673a198436af10738b7

File: dm/tests/check_task/run.sh [hunk 1]
```diff
@@ -36,7 +30,10 @@ function test_check_task_fail_no_block() {
...
 
 function test_check_task_fail_no_block_forsharding() {
-	prepare_many_tables
+	run_sql_both_source "drop database if exists \`check-task\`"
+	run_sql_both_source "create database if not exists \`check-task\`"
+	run_sql_file $cur/data/db1.prepare.sql $MYSQL_HOST1 $MYSQL_PORT1 $MYSQL_PASSWORD1
+
 	run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
 		"check-task $cur/conf/task-sharding.yaml" \
 		"\"state\": \"fail\"" 1
```
Comment (lance6716) [inline dm/tests/check_task/run.sh:39]: cleanup_data some lines below can also clean `check-task`

rest lgtm

Comment (Ehco1996) [reply dm/tests/check_task/run.sh:39]: fixed in https://github.com/pingcap/tiflow/commit/3c0284bcea3647d2681524e814439ef4b0694752

Comment (lance6716) [reply dm/tests/check_task/run.sh:39]: I mean cleanup_data checktask check-task in line 50, 54

You change will not be executed if test fails before it and will not clean the environment before `run`

Comment (Ehco1996) [reply dm/tests/check_task/run.sh:39]: https://github.com/pingcap/tiflow/pull/5896/commits/42a58a2a63f6ee50d297efe16babb2dc57f7cd0b 😬

**Review by lance6716 — APPROVED** [review]

**Comment by lance6716** [comment]
```
[2022-06-18T03:19:16.780Z] [2022/06/18 11:18:35.012 +08:00] [ERROR] [baseconn.go:184] ["execute statement failed"] [task=test] [unit="binlog replication"] [query="ALTER TABLE `shardddl`.`tb` MODIFY COLUMN `a` DOUBLE"] [argument="[]"] [error="Error 8245: column a id 2 does not exist, this column may have been updated by other DDL ran in parallel"]
```

waiting https://github.com/pingcap/tiflow/pull/5916

please take a look at CI before bindly retry

**Pull request merged**